core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala

```diff
@@ -464,6 +464,31 @@ abstract class Stream[+T] { self =>
       v
     }
   }
+
+  def withFallBack(fallBack: Stream[T @uncheckedVariance],
+                   printOnFallback: Boolean = false): Stream[T] = {
+    var defaultValueIsFresh = true
+
+    val cancel1 = fallBack.foreach(_ => defaultValueIsFresh = false)
```

Comment:
As soon as this method returns, these variables are no longer needed and may be cleaned up. You should probably create a new class for a fallback stream that stores these listeners.

This is super janky, but will be fixed with the new stream tracking mechanism, which I hope to PRify this week.

Reply:
Wait, but they are needed though. It's to make sure that the foreach stuff is never cancelled

Reply:
I mean from the JVM perspective, those variables are not depended on by anyone and so would be cleaned out.

Reply:
Would that mean that 

https://github.com/Team846/potassium/blob/198a1f43c570bf9ca8dc4e7a8f1a50eee685cc94/core/shared/src/main/scala/com/lynbrookrobotics/potassium/streams/Stream.scala#L510
 
will break?